src/SSHDebugPS/Docker/DockerHelper.cs

```diff
@@ -10,13 +10,111 @@
 using Microsoft.SSHDebugPS.SSH;
 using Microsoft.SSHDebugPS.Utilities;
 
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+
 namespace Microsoft.SSHDebugPS.Docker
 {
     public class DockerHelper
     {
         private const string dockerPSCommand = "ps";
         // --no-trunc avoids parameter truncation
         private const string dockerPSArgs = "-f status=running --no-trunc --format \"{{json .}}\"";
+
+        public static string GetDockerOutputString(string hostname, string dockerCommand, string dockerArgs)
+        {
+            string path = null;
+            try
+            {
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
+                {
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                    path = GetDockerOutputStringImpl(hostname, dockerCommand, dockerArgs);
+                });
+            }
+            catch (Exception)
+            {
```

Comment:
A blanket catch statement shouldn't be done here. You should handle `CommandFailedException` messages separately and then write an error.